src/config.rs

```diff
@@ -16,16 +16,23 @@ use directories::ProjectDirs;
 use regex::Regex;
 use serde::de::DeserializeOwned;
 use serde_json;
+use std::collections::HashMap;
 use std::env;
-use std::io::Read;
-use std::fs::File;
+use std::io::{Read, Write};
+use std::fs::{self, File};
 use std::net::IpAddr;
 use std::path::{Path, PathBuf};
 use std::str::FromStr;
+use std::sync::Mutex;
 use toml;
 
+use errors::*;
+
 lazy_static! {
-    pub static ref CONFIG: Config = { Config::create() };
+    pub static ref CONFIG: Config = Config::create();
+
+    static ref CACHED_CONFIG_PATH: PathBuf = CachedConfig::file_config_path();
```

Comment:
I'm not wild about "cached" as the naming here, especially since we have a lot of other usage of "cache" in the codebase. We could just directly name it "OauthTokens" or something like that.